File: src/main/java/com/lessmarkup/userinterface/model/structure/NavigationBarModel.java
package com.lessmarkup.userinterface.model.structure;

import java.util.ArrayList;
import java.util.List;

public class NavigationBarModel extends MenuItemModel {
    private final List<NavigationBarModel> children = new ArrayList<>();

    public NavigationBarModel(String url, String title, String imageUrl, boolean selected, int level) {
        super(url, title, imageUrl, selected, level);
    }

    public List<NavigationBarModel> getChildren() { return children; }
}

File: src/main/scala/com/lessmarkup/userinterface/model/structure/NavigationBarModel.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
 * If a copy of the MPL was not distributed with this file, You can obtain one at
 * http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.userinterface.model.structure

class NavigationBarModel(url: String, title: String, imageUrl: String, selected: Boolean, level: Int) extends MenuItemModel(url, title, imageUrl, selected, level) {

  private final val children: List[NavigationBarModel] = List()

  def getChildren: List[NavigationBarModel] = children
}
